src/backend/routers/chat.py

```diff
@@ -49,57 +45,47 @@ async def chat_stream(
         session (DBSessionDep): Database session.
         chat_request (CohereChatRequest): Chat request data.
         request (Request): Request object.
+        ctx (Context): Context object.
 
     Returns:
         EventSourceResponse: Server-sent event response with chatbot responses.
     """
-    trace_id = None
-    if hasattr(request.state, "trace_id"):
-        trace_id = request.state.trace_id
-    add_model_to_request_state(request, chat_request.model)
-    user_id = request.headers.get("User-Id", None)
+    ctx.with_model(chat_request.model)
+
     agent_id = chat_request.agent_id
+    ctx.with_agent_id(agent_id)
+
+    agent = DEFAULT_METRICS_AGENT
     if agent_id:
         agent = agent_crud.get_agent_by_id(session, agent_id)
-        add_agent_to_request_state(request, agent)
-    else:
-        add_default_agent_to_request_state(request)
+    ctx.with_agent(agent)
```

Comment:
`Agent` and `MetricsAgent` are different types and don't have a subtype r/s; using them in this way can lead to bugs